### PR TITLE
Silence `console.log`s

### PR DESCRIPTION
**Pull request by trond-snekvik**
Turns the `console.log` messages that happen regularly into output stream log messages. Leaves `console.log`s in exception handlers the way they are.

`console.log` will show up in the console window of all extension debug sessions. I'm the maintainer of [an extension](https://marketplace.visualstudio.com/items?itemName=nordic-semiconductor.nrf-connect) that integrates with the configuration provider API for the C++ extension. 

The amount of console log messages this extension generates that I have to filter out when debugging my own extension finally got annoying enough that I wanted to open a PR to try and force a change.

Note that I'm bypassing the localization mechanism for the output channel here. Not sure if this is acceptable.

Fixes most of #8389, but there are still a few `console.log`s left, so not sure if it's enough to mark it as closed.

### CODE REVIEW

**Review by sean-mcmanus — APPROVED** [review]

File: Extension/src/LanguageServer/client.ts [hunk 1]
```diff
@@ -2239,7 +2239,7 @@ export class DefaultClient implements Client {
                 try {
                     promise.resolve(task ? await task() : undefined);
                 } catch (e) {
-                    console.log(e);
+                    log(e);
```
Comment (sean-mcmanus) [inline Extension/src/LanguageServer/client.ts]: The compiler wants this `e` to be a string -- maybe keep this as console.log?

File: Extension/src/LanguageServer/client.ts [hunk 2]
```diff
@@ -3072,32 +3072,32 @@ export class DefaultClient implements Client {
...
+                    log("Received an invalid browse configuration from configuration provider.");
                 }
                 const configValue: WorkspaceBrowseConfiguration | undefined = this.lastCustomBrowseConfiguration.Value;
                 if (configValue) {
                     sanitized = configValue;
                     if (sanitized.browsePath.length === 0) {
                         sanitized.browsePath = ["${workspaceFolder}/**"];
                     }
-                    console.log("Falling back to last received browse configuration: ", JSON.stringify(sanitized, null, 2));
+                    log("Falling back to last received browse configuration: ", JSON.stringify(sanitized, null, 2));
```
Comment (sean-mcmanus) [inline Extension/src/LanguageServer/client.ts]: `, ` should be ` + `

File: Extension/src/LanguageServer/client.ts [hunk 3]
```diff
@@ -3072,32 +3072,32 @@ export class DefaultClient implements Client {
...
-                console.log("Received an invalid browse configuration from configuration provider: " + JSON.stringify(sanitized));
+                log("Received an invalid browse configuration from configuration provider: " + JSON.stringify(sanitized));
                 const configValue: WorkspaceBrowseConfiguration | undefined = this.lastCustomBrowseConfiguration.Value;
                 if (configValue) {
                     sanitized = configValue;
                     if (sanitized.browsePath.length === 0) {
                         sanitized.browsePath = ["${workspaceFolder}/**"];
                     }
-                    console.log("Falling back to last received browse configuration: ", JSON.stringify(sanitized, null, 2));
+                    log("Falling back to last received browse configuration: ", JSON.stringify(sanitized, null, 2));
```
Comment (sean-mcmanus) [inline Extension/src/LanguageServer/client.ts]: `, ` should be ` + `

**Review by sean-mcmanus — CHANGES_REQUESTED** [review]

**Comment by bobbrow** [comment]
I'm honestly not sure how critical it is to keep any/most of these messages.  I added them years ago as a cue to developers working on config providers.  Since average users generally don't need to see these, they should be logged with a higher level to be hidden by default, or simply deleted if we don't think they add value.

**Comment by sean-mcmanus** [comment]
> I'm honestly not sure how critical it is to keep any/most of these messages. I added them years ago as a cue to developers working on config providers. Since average users generally don't need to see these, they should be logged with a higher level to be hidden by default, or simply deleted if we don't think they add value.

i.e. it sounds like some logDebug or equivalent should be used instead of log, so that it doesn't log these messages unless the logging_level is "Debug".

**Comment by bobbrow** [comment]
> i.e. it sounds like some logDebug or equivalent should be used instead of log, so that it doesn't log these messages unless the logging_level is "Debug".

In case my last comment was unclear, I'm totally ok with deleting all of these messages.

**Comment by sean-mcmanus** [comment]
> > i.e. it sounds like some logDebug or equivalent should be used instead of log, so that it doesn't log these messages unless the logging_level is "Debug".
> 
> In case my last comment was unclear, I'm totally ok with deleting all of these messages.

Sure, but it seems like `Received an invalid browse configuration from configuration provider: ` might be useful to log.

File: Extension/src/LanguageServer/client.ts (hunk 1 above)
Comment (trond-snekvik) [reply Extension/src/LanguageServer/client.ts]: Kept this as-is.

File: Extension/src/LanguageServer/client.ts (hunk 2 above)
Comment (trond-snekvik) [reply Extension/src/LanguageServer/client.ts]: Removed this line altogether

File: Extension/src/LanguageServer/client.ts (hunk 3 above)
Comment (trond-snekvik) [reply Extension/src/LanguageServer/client.ts]: Removed this line altogether

**Comment by trond-snekvik** [comment]
Thanks for the feedback. As suggested, I removed the logs instead of moving them to the output channel. Kept  the ones that report irregular behavior.

**Review by bobbrow — APPROVED** [review]

**Review by sean-mcmanus — APPROVED** [review]

**Pull request merged**